lib/contentTypeParser.js

```diff
@@ -358,63 +348,15 @@ function removeAllContentTypeParsers () {
   this[kContentTypeParser].removeAll()
 }
 
-// dummy here to prevent repeated object creation
-const dummyContentType = { type: '', parameters: Object.create(null) }
-
-function safeParseContentType (contentType) {
-  try {
-    return parseContentType(contentType)
-  } catch (err) {
-    return dummyContentType
-  }
-}
-
-function compareContentType (contentType, parserListItem) {
-  if (parserListItem.isEssence) {
-    // we do essence check
-    return contentType.type.indexOf(parserListItem) !== -1
-  } else {
-    // when the content-type includes parameters
-    // we do a full-text search
-    // reject essence content-type before checking parameters
-    if (contentType.type.indexOf(parserListItem.type) === -1) return false
-    for (const key of parserListItem.parameterKeys) {
-      // reject when missing parameters
-      if (!(key in contentType.parameters)) return false
-      // reject when parameters do not match
-      if (contentType.parameters[key] !== parserListItem.parameters[key]) return false
-    }
-    return true
+function validateRegExp (regexp) {
+  // RegExp should either start with ^ or include ;?
+  // It can ensure the user is properly detect the essence
+  // MIME types.
+  if (regexp.source[0] !== '^' && !regexp.source.includes(';?')) {
```

Comment:
```suggestion
  if (regexp.source[0] !== '^' && regexp.source.includes(';?') === true) {
```

Reply:
Should be `=== false`.

```suggestion
  if (regexp.source[0] !== '^' && regexp.source.includes(';?') === false) {
```